src/plugins/intel_gpu/src/kernel_selector/kernels/gather/gather_kernel_ref.cpp

```diff
@@ -211,11 +236,15 @@ CommonDispatchData GatherKernelRef::SetDefault(const gather_params& params) cons
 JitConstants GatherKernelRef::GetJitConstants(const gather_params& params) const {
     JitConstants jit = MakeBaseParamsJitConstants(params);
 
+    jit.AddConstant(MakeJitConstant("GATHER_AXIS_SHAPE_INFO_INDEX", GetGatherAxisIndexInShapeInfo(params)));
```

Comment:
I think we need to add this only when needed, i.e., params.is_shape_agnostic

Reply:
done